CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -307,65 +328,149 @@ void SiStripGainFromCalibTree::algoBeginRun(const edm::Run& run, const edm::Even
+              APV->y             = DetUnit->position().basicVector().y();
+              APV->z             = DetUnit->position().basicVector().z();
+              APV->Eta           = DetUnit->position().basicVector().eta();
+              APV->Phi           = DetUnit->position().basicVector().phi();
+              APV->R             = DetUnit->position().basicVector().transverse();
+              APV->Thickness     = DetUnit->surface().bounds().thickness();
+              APV->isMasked      = false; //SiPixelQuality_->IsModuleBad(Detid.rawId());
+              APV->NEntries      = 0;
+
+              APVsCollOrdered.push_back(APV);
+              APVsColl[(APV->DetId<<4) | APV->APVId] = APV;
+              Index++;
+              NPixelDets++;
+           }}
+      }
+   }
+
+
    MakeCalibrationMap();
 
    NEvent     = 0;
    NTrack     = 0;
-   NCluster   = 0;
+   NClusterStrip   = 0;
+   NClusterPixel   = 0;
    SRun       = 1<<31;
    ERun       = 0;
    GOOD       = 0;
    BAD        = 0;
    MASKED     = 0;
-   
-   cout << "algoBeginRun end" << endl;
 }
 
+
+
+void SiStripGainFromCalibTree::algoBeginRun(const edm::Run& run, const edm::EventSetup& iSetup)
+{
+   edm::ESHandle<SiStripGain> gainHandle;
+   iSetup.get<SiStripGainRcd>().get(gainHandle);
+   if(!gainHandle.isValid()){edm::LogError("SiStripGainFromCalibTree")<< "gainHandle is not valid\n"; exit(0);}
+ 
+   edm::ESHandle<SiStripQuality> SiStripQuality_;
+   iSetup.get<SiStripQualityRcd>().get(SiStripQuality_);
+
+   for(unsigned int a=0;a<APVsCollOrdered.size();a++){
+      stAPVGain* APV = APVsCollOrdered[a];
+
+      APV->isMasked      = SiStripQuality_->IsApvBad(APV->DetId,APV->APVId);
+//      if(!FirstSetOfConstants){
+         if(gainHandle->getNumberOfTags()!=2){edm::LogError("SiStripGainFromCalibTree")<< "NUMBER OF GAIN TAG IS EXPECTED TO BE 2\n";fflush(stdout);exit(0);};		   
+         float newPreviousGain = gainHandle->getApvGain(APV->APVId,gainHandle->getRange(APV->DetId, 1),1);
+         if(APV->PreviousGain!=1 and newPreviousGain!=APV->PreviousGain)edm::LogWarning("SiStripGainFromCalibTree")<< "WARNING: ParticleGain in the global tag changed\n";
+         APV->PreviousGain = newPreviousGain;
+
+         float newPreviousGainTick = gainHandle->getApvGain(APV->APVId,gainHandle->getRange(APV->DetId, 0),1);
+         if(APV->PreviousGainTick!=1 and newPreviousGainTick!=APV->PreviousGainTick)edm::LogWarning("SiStripGainFromCalibTree")<< "WARNING: TickMarkGain in the global tag changed\n";
+         APV->PreviousGainTick = newPreviousGainTick;
+
+
+         //printf("DETID = %7i APVID=%1i Previous Gain=%8.4f (G1) x %8.4f (G2)\n",APV->DetId,APV->APVId,APV->PreviousGainTick, APV->PreviousGain);
+         
+         
+
+//      }
+   }
+}
+
+void SiStripGainFromCalibTree::algoEndRun(const edm::Run& run, const edm::EventSetup& iSetup){
+   if(AlgoMode == "PCL" && !harvestingMode)return;//nothing to do in that case
+
+   if(AlgoMode == "PCL" and harvestingMode){
```

Comment:
Seems not to be commonly used. I don't have a strong opinion (as opposed to some others where there are clear bugs)



On Sep 4, 2015, at 2:02 PM, Chris Jones <notifications@github.com<mailto:notifications@github.com>> wrote:


In CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc<https://github.com/cms-sw/cmssw/pull/10680#discussion_r38744516>:

> +         if(APV->PreviousGainTick!=1 and newPreviousGainTick!=APV->PreviousGainTick)edm::LogWarning("SiStripGainFromCalibTree")<< "WARNING: TickMarkGain in the global tag changed\n";
> +         APV->PreviousGainTick = newPreviousGainTick;
> +
> +
> +         //printf("DETID = %7i APVID=%1i Previous Gain=%8.4f (G1) x %8.4f (G2)\n",APV->DetId,APV->APVId,APV->PreviousGainTick, APV->PreviousGain);
> +
> +
> +
> +//      }
> +   }
> +}
> +
> +void SiStripGainFromCalibTree::algoEndRun(const edm::Run& run, const edm::EventSetup& iSetup){
> +   if(AlgoMode == "PCL" && !harvestingMode)return;//nothing to do in that case
> +
> +   if(AlgoMode == "PCL" and harvestingMode){


Out of Curiosity what's the motivation? I personally prefer and.

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/10680/files#r38744516>.